src/LightBulbAccessory.ts

```diff
@@ -71,30 +74,38 @@ export default class LightBulbAccessory {
       .onGet(Brightness.get.bind(this))
       .onSet(Brightness.set.bind(this));
 
-    this.service
-      .getCharacteristic(this.platform.Characteristic.ColorTemperature)
-      .setProps({
-        minValue: HOME_KIT_VALUES.min,
-        maxValue: HOME_KIT_VALUES.max
-      })
-      .onGet(ColorTemperature.get.bind(this))
-      .onSet(ColorTemperature.set.bind(this));
+    if (supportsColorTemp == true) {
+      this.service
+        .getCharacteristic(this.platform.Characteristic.ColorTemperature)
+        .setProps({
+          minValue: HOME_KIT_VALUES.min,
+          maxValue: HOME_KIT_VALUES.max
+        })
+        .onGet(ColorTemperature.get.bind(this))
+        .onSet(ColorTemperature.set.bind(this));
+    }
 
-    this.service
-      .getCharacteristic(this.platform.Characteristic.Hue)
-      .onGet(Hue.get.bind(this))
-      .onSet(Hue.set.bind(this));
+    if (supportsHue == true) {
+      this.service
+        .getCharacteristic(this.platform.Characteristic.Hue)
+        .onGet(Hue.get.bind(this))
+        .onSet(Hue.set.bind(this));
+    }
 
-    this.service
-      .getCharacteristic(this.platform.Characteristic.Saturation)
-      .onGet(Saturation.get.bind(this))
-      .onSet(Saturation.set.bind(this));
-
-    const adaptiveLightingController =
-      new this.platform.api.hap.AdaptiveLightingController(this.service, {
-        controllerMode:
-          this.platform.api.hap.AdaptiveLightingControllerMode.AUTOMATIC
-      });
+    if (supportsSaturation == true) {
+      this.service
+        .getCharacteristic(this.platform.Characteristic.Saturation)
+        .onGet(Saturation.get.bind(this))
+        .onSet(Saturation.set.bind(this));
+    }
+
+    if (supportsColorTemp == true && supportsHue == true && supportsSaturation == true) {
+      const adaptiveLightingController =
+        new this.platform.api.hap.AdaptiveLightingController(this.service, {
+          controllerMode:
+            this.platform.api.hap.AdaptiveLightingControllerMode.AUTOMATIC
+        });
+    }
 
     this.accessory.configureController(adaptiveLightingController);
```

Comment:
You missed this line of code, move this inside the if, and please tes this logic, by running a local homebridge